Release/1.3.0 (#4211)
aiidateam/aiida-core: CHANGELOG.md

# Changelog

## v1.3.0

### Improvements
- Comprehensive restructuring and revamp of the online documentation [[#4141]](https://github.com/aiidateam/aiida-core/pull/4141)

> - Improving SSH configuration defaults [[#4055]](https://github.com/aiidateam/aiida-core/pull/4055)

Reviewer: ```suggestion
- Improving defaults for `verdi computer configure ssh` [[#4055]](https://github.com/aiidateam/aiida-core/pull/4055)
```

### Performance

> - Reinstate broadcast subscription of workchain for child processes [[#4154]](https://github.com/aiidateam/aiida-core/pull/4154)

Reviewer: ```suggestion
- Enable event-based monitoring of Workchain child processes (child processes were being polled every second) ( [[#4154]](https://github.com/aiidateam/aiida-core/pull/4154)
```

> - Increase the default for `runner.poll.interval` config option to 60 [[#4150]](https://github.com/aiidateam/aiida-core/pull/4150)

Reviewer: ```suggestion
- Increase the default for `runner.poll.interval` config option from 1 to 60 seconds [[#4150]](https://github.com/aiidateam/aiida-core/pull/4150)
```

- Increase the efficiency of the `SqlaGroup.nodes` iterator [[#4094]](https://github.com/aiidateam/aiida-core/pull/4094)

### Features
- Add a progress bar for export and import related functionality [[#3599]](https://github.com/aiidateam/aiida-core/pull/3599)

> - Enable loading config.yml files from URL for ceratin `verdi` commands [[#3977]](https://github.com/aiidateam/aiida-core/pull/3977)

Reviewer: ```suggestion
- Enable loading config.yml files from URL in `verdi` commands with `--config` option [[#3977]](https://github.com/aiidateam/aiida-core/pull/3977)
```

- `QueryBuilder`: add the `flat` argument to the `.all()` method [[#3945]](https://github.com/aiidateam/aiida-core/pull/3945)
- `verdi status`: add `--no-rmq` flag to skip the RabbitMQ check [[#4181]](https://github.com/aiidateam/aiida-core/pull/4181)
- Add support for process functions in `verdi plugin list` [[#4117]](https://github.com/aiidateam/aiida-core/pull/4117)

> - Allow profile selection ipython magic %aiida [[#4071]](https://github.com/aiidateam/aiida-core/pull/4071)

Reviewer: ```suggestion
- Allow profile selection in ipython magic %aiida [[#4071]](https://github.com/aiidateam/aiida-core/pull/4071)
```

- Support more complex formula formats in `aiida.orm.data.cif.parse_formula` [[#3954]](https://github.com/aiidateam/aiida-core/pull/3954)

### Bug fixes
- `BaseRestartWorkChain`: do not assume `metadata` exists in inputs in `run_process` [[#4210]](https://github.com/aiidateam/aiida-core/pull/4210)
- `BaseRestartWorkChain`: fix bug in `inspect_process` [[#4166]](https://github.com/aiidateam/aiida-core/pull/4166)

> - `BaseRestartWorkChain`: fix the unhandled failure mechanism [[#4155]](https://github.com/aiidateam/aiida-core/pull/4155)

Reviewer: ```suggestion
- `BaseRestartWorkChain`: fix the "unhandled failure mechanism" for dealing with failures of subprocesses [[#4155]](https://github.com/aiidateam/aiida-core/pull/4155)
```

- `CalcJob`: move job resource validation to the `Scheduler` class fixing a problem for the SGE and LSF scheduler plugins [[#4192]](https://github.com/aiidateam/aiida-core/pull/4192)
- `WorkChain`: guarantee to maintain order of appended awaitables [[#4156]](https://github.com/aiidateam/aiida-core/pull/4156)
- Add support for binary files to the various `verdi` cat commands [[#4077]](https://github.com/aiidateam/aiida-core/pull/4077)
- Ensure `verdi group show --limit` respects limit even in raw mode [[#4092]](https://github.com/aiidateam/aiida-core/pull/4092)
- `QueryBuilder`: fix type string filter generation for `Group` subclasses [[#4144]](https://github.com/aiidateam/aiida-core/pull/4144)
- Raise when calling `Node.objects.delete` for node with incoming links [[#4168]](https://github.com/aiidateam/aiida-core/pull/4168)
- Properly handle multiple requests to threaded REST API [[#3974]](https://github.com/aiidateam/aiida-core/pull/3974)
- `NodeTranslator`: do not assume `get_export_formats` exists [[#4188]](https://github.com/aiidateam/aiida-core/pull/4188)

> - Reinstate highlighting of selected nodes in generated images of provenance graph [[#4081]](https://github.com/aiidateam/aiida-core/pull/4081)

Reviewer: ```suggestion
- Provenance graphs: enable highlighting specific node classes (and highlight root node by default) [[#4081]](https://github.com/aiidateam/aiida-core/pull/4081)
```
This could also be moved under "Improvements"
